nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -440,10 +446,16 @@ sub handle_modified_unit { ## no critic(Subroutines::ProhibitManyArgs, Subroutin
         # properties (resource limits and inotify watches)
         # seem to get applied on daemon-reload.
     } elsif ($unit =~ /\.mount$/msx) {
-        # Reload the changed mount unit to force a remount.
-        # FIXME: only reload when Options= changed, restart otherwise
-        $units_to_reload->{$unit} = 1;
-        record_unit($reload_list_file, $unit);
+        # Just restart the unit. We wouldn't have gotten into this subroutine
+        # if only `Options` was changed, in which case the unit would be reloaded.
+        # The only exception is / and /nix because it's very unlikely we can safely
+        # unmount them.
+        if ($unit eq "-.mount" or $unit eq "nix.mount") {
+            $units_to_skip->{$unit} = 1;
```

Comment:
This will now skip reloading `/` and `/nix` right? Previously it would've been reloaded. Is that intended?

Reply:
Well yes but actually no ;) It's another part in the spghetti. This is `handle_modified_unit` which is only called when `compare_units` didn't return 2. My change in that subroutine causes it to return 2 which mean that `handle_modified_unit` isn't called in the reload case anymore

Reply:
Ah I see what you mean